lib/ansible/modules/cloud/amazon/cloudtrail.py

```diff
@@ -19,231 +19,583 @@
-        if len(trailList) == 1:
-            return trailList[0]
+            client.stop_logging(Name=name)
+            return client.get_trail_status(Name=name)
+        except ClientError as err:
+            module.fail_json(msg=str(err))
+    else:
+        module.fail_json(msg="Unsupported logging action")
+
+
+def get_trail_facts(module, client, name):
+    """
+    Describes existing trail in an account
+
+    module : AnsibleModule object
+    client : boto3 client connection object
+    name : Name of the trail
+    """
+    # get Trail info
+    try:
+        trail_resp = client.describe_trails(trailNameList=[name])
+    except ClientError as err:
+        module.fail_json(msg=str(err))
+
+    # Now check to see if our trail exists and get status and tags
+    if len(trail_resp['trailList']):
+        trail = trail_resp['trailList'][0]
+        try:
+            status_resp = client.get_trail_status(Name=trail['Name'])
+            tags_list = client.list_tags(ResourceIdList=[trail['TrailARN']])
+        except ClientError as err:
+            module.fail_json(msg=str(err))
+
+        trail['IsLogging'] = status_resp['IsLogging']
+        trail['tags'] = boto3_tag_list_to_ansible_dict(tags_list['ResourceTagList'][0]['TagsList'])
+        # Check for non-existent values and populate with None
+        optional_vals = set(['S3KeyPrefix', 'SnsTopicName', 'SnsTopicARN', 'CloudWatchLogsLogGroupArn', 'CloudWatchLogsRoleArn', 'KmsKeyId'])
+        for v in optional_vals - set(trail.keys()):
+            trail[v] = None
+        return trail
+
+    else:
+        # trail doesn't exist return None
         return None
 
-    def exists(self, name=None):
-        ret = self.view(name)
-        if ret:
-            return True
-        return False
-
-    def enable_logging(self, name):
-        '''Turn on logging for a cloudtrail that already exists. Throws Exception on error.'''
-        self.conn.start_logging(name)
 
+def delete_trail(module, client, trail_arn):
+    """
+    Delete a CloudTrail
 
-    def enable(self, **create_args):
-        return self.conn.create_trail(**create_args)
+    module : AnisbleModule object
+    client : boto3 client connection object
+    trail_arn : Full CloudTrail ARN
+    """
+    try:
+        client.delete_trail(Name=trail_arn)
+    except ClientError as err:
+        module.fail_json(msg=str(err))
 
-    def update(self, **create_args):
-        return self.conn.update_trail(**create_args)
 
-    def delete(self, name):
-        '''Delete a given cloudtrial configuration. Throws Exception on error.'''
-        self.conn.delete_trail(name)
+def update_trail(module, client, ct_params):
+    """
+    Delete a CloudTrail
 
+    module : AnisbleModule object
+    client : boto3 client connection object
+    ct_params : The parameters for the Trail to update
+    """
+    try:
+        client.update_trail(**ct_params)
+    except ClientError as err:
+        module.fail_json(msg=str(err))
```

Comment:
Use standard boto3 exception handling: 

- https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-2

I've just noticed there's a missing `c` at the end of `traceback.format_exc()`, so just be careful  there :)